src/details/ArborX_DetailsDistributedTreeImpl.hpp

```diff
@@ -417,6 +449,9 @@ DistributedTreeImpl<DeviceType>::queryDispatchImpl(
   // recompute everything instead of just searching for potential better
   // neighbors and updating the list.
 
+  CallbackWithDistance<BVH<typename DeviceType::memory_space>>
```

Comment:
Not sure if I should try to fix it to something like `std::decay_t<decltype(bottom_tree)>`. It is still completely tied to `BVH` and won't work with anything else.

Reply:
Just add a comment before you merge